cdap-app-fabric/src/main/java/co/cask/cdap/security/DefaultUGIProvider.java

```diff
@@ -87,6 +88,8 @@ public UserGroupInformation getConfiguredUGI(ImpersonationInfo impersonationInfo
       String expandedPrincipal = SecurityUtil.expandPrincipal(impersonationInfo.getPrincipal());
       LOG.debug("Logging in as: principal={}, keytab={}", expandedPrincipal, localKeytabFile);
 
+      FileUtils.assertReadableFile(localKeytabFile);
```

Comment:
You might want to check if it is `isRegularFile`. Also, use `Files` method so that you can follow symbolic link.